source/input/InputSource.cpp

```diff
@@ -14,4 +17,105 @@ void InputSource::removeDispatcher(EventDispatcher &d) {
   }), dispatchers.end());
 }
 
+std::string InputSource::mouseButtonToString(const int &mouseButton) {
+  switch ((MouseButton)mouseButton) {
+    case MouseButton::MOUSE_BUTTON_LEFT:    return "mouse_left";
+    case MouseButton::MOUSE_BUTTON_RIGHT:   return "mouse_right";
+    case MouseButton::MOUSE_BUTTON_MIDDLE:  return "mouse_middle";
+    default:                                return "";
+  }
+}
+
+std::string InputSource::mouseAxisToString(const int &mouseAxis) {
+  switch ((MouseAxis)mouseAxis) {
+    case MouseAxis::MOUSE_AXIS_X: return "mouse_x";
+    case MouseAxis::MOUSE_AXIS_Y: return "mouse_y";
+    default:                      return "";
+  }
+}
+
+int InputSource::keyboardGetKeyFromString(const std::string &key) {
+	return (int)SDL_GetScancodeFromName(key.c_str());
+}
+
+int InputSource::mouseGetButtonFromString(const std::string &buttonStr) {
```

Comment:
Could all of this be done with a lookup table? It would look a bit nicer : )

Reply:
@Geertiebear Should this be fixed for this PR to be approved or can this be a new issue?

Reply:
It was a point of dicussion, but if it is to be done I think it should be done before the PR is merged.

Reply:
@babyccino

Reply:
Yeah I can do this tonight. I want to try and make InputManager use event observers as well

Reply:
@babyccino Are you blocked by anything? If so I'll see that I get you unblocked.

Reply:
Sorry have been busy due to renovations and such. Not blocked as far as I can tell.

Reply:
:+1: :+1: :+1: 
Nice.